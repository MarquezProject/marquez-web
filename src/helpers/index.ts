import { IDatasetAPI, IJobAPI, INetworkData, INodeNetwork } from '../types/api'

export const createRollbarMessage = (
  functionName: string,
  e: string,
  severity: 'critical' | 'error' | 'warning' | 'info' | 'debug' = 'error'
) => {
  if (__NODE_ENV__ === 'production') {
    if (__ROLLBAR__) {
      Rollbar[severity](`Error in ${functionName}: ${e}`)
    }
  }
}

export const createNetworkData = (datasets: IDatasetAPI[], jobs: IJobAPI[]): INetworkData => {
  console.log('jobs', jobs.length)
  console.log('What to do if there are matching jobs but no matching datasets?')
  const datasetNodes: INodeNetwork[] = datasets.map(d => ({
    id: d.name,
    tag: 'dataset'
  }))

  const jobNodes: INodeNetwork[] = jobs.map(j => ({
    id: j.name,
    tag: 'job'
  }))

  const links = jobs.reduce((links, singleJob) => {
    const inLinks = singleJob.inputs.map(input => ({
      source: input,
      target: singleJob.name
    }))
    const outLinks = singleJob.outputs.map(output => ({
      source: singleJob.name,
      target: output
    }))

    return [...links, ...inLinks, ...outLinks]
  }, [])
  return {
    nodes: [...datasetNodes, ...jobNodes],
    links
  }
}
