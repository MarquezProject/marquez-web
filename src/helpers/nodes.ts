import { LineageDataset, LineageJob, MqNode } from '../components/lineage/types'
import { Undefinable } from '../types/util/Nullable'

export function isJob(node: MqNode): Undefinable<LineageJob> {
  if (node.data.type === 'BATCH') {
    return node.data as LineageJob
  }
  return undefined
}

export function isDataset(node: MqNode): Undefinable<LineageDataset> {
  if (node.data.type === 'DB_TABLE') {
    return node.data as LineageDataset
  }
  return undefined
}
