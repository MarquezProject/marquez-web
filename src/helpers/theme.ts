
import {createMuiTheme} from '@material-ui/core/styles'

export const theme = createMuiTheme({
  overrides: {
    MuiCssBaseline: {
      '@global': {
        body: {
          color: '#fff'
        }
      },
    },
  },
  typography: {
    h3: {
      fontSize: '1rem',
      fontWeight: 700,
      lineHeight: 2,
    },
    fontSize: 14
  },
  palette: {
    type: 'dark',
    primary: {
      main: '#3587e8'
    },
    background: {
      default: '#191f26'
    },
    secondary: {
      main: '#454f5b'
    }
  }
})

// 1px for bottom border
export const HEADER_HEIGHT = 96 + 1
