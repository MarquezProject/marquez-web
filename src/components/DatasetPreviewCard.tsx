import React, { ReactElement } from 'react'
import {
  withStyles,
  createStyles,
  WithStyles as IWithStyles,
  Theme as ITheme
} from '@material-ui/core/styles'
import { Typography, Box } from '@material-ui/core'
import { formatUpdatedAt } from '../helpers'
import tagToBadge from '../config/tag-to-badge'

import { IDatasetAPI } from '../types/api'

const styles = ({ palette }: ITheme) => {
  return createStyles({
    rightCol: {
      textAlign: 'right'
    },
    lastUpdated: {
      color: palette.grey[600]
    }
  })
}

const StyledTypography = withStyles({
  root: {
    maxWidth: '90%'
  }
})(Typography)

type IProps = IWithStyles<typeof styles> &
  Pick<IDatasetAPI, 'name' | 'description' | 'updatedAt' | 'tags'>
interface IState {}

class DatasetPreviewCard extends React.Component<IProps, IState> {
  render(): ReactElement {
    const { classes, name, description, updatedAt = 'error', tags = ['is_pii'] } = this.props
    return (
      <Box p={2} m={1} bgcolor='white' boxShadow={3} display='flex' justifyContent='space-between'>
        <div>
          <Typography color='secondary' variant='h3'>
            {name}
          </Typography>
          <StyledTypography color='primary'>{description}</StyledTypography>
        </div>
        <Box
          className={classes.rightCol}
          display='flex'
          flexDirection='column'
          alignItems='flex-end'
          justifyContent='space-between'
        >
          <div id='tagContainer'>{tags.map(t => tagToBadge[t])}</div>
          <Typography className={classes.lastUpdated}>{formatUpdatedAt(updatedAt)}</Typography>
        </Box>
      </Box>
    )
  }
}

export default withStyles(styles)(DatasetPreviewCard)
