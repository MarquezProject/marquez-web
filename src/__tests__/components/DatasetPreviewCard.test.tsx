import { mount } from 'enzyme'
import * as React from 'react'
import Typography from '@material-ui/core/Typography'
import DatasetPreviewCard from '../../components/DatasetPreviewCard'
import { formatUpdatedAt } from '../../helpers'
import { MemoryRouter } from 'react-router-dom'

const datasets = require('../../../docker/db/data/datasets.json')
const dataset = datasets[0]
const tags = ['TAG_A', 'TAG_B', 'TAG_C']
dataset.tags = tags

describe('DatasetPreviewCard Component', () => {
  const wrapper = mount(<MemoryRouter><DatasetPreviewCard { ...dataset }/></MemoryRouter>)
  it('Should render', () => {
    expect(wrapper.exists()).toBe(true)
  })
  
  const componentText = wrapper.render().text()

  it('should render the dataset name', () => {
    expect(componentText).toContain(dataset.name)
  })
  it('should render the description', () => {
    expect(componentText).toContain(dataset.description)
  })
  it('should render the time', () => {
    expect(
      wrapper
        .find(Typography)
        .last()
        .text()
    ).toContain(formatUpdatedAt(dataset.updatedAt))
  })
  // wrapping in Router produces a new key each time, which makes the snapshots not match
  test.skip('renders a snapshot that matches previous', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
