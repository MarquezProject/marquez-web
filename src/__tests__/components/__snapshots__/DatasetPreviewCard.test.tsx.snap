// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetPreviewCard Component renders a snapshot that matches previous 1`] = `
<WithStyles(DatasetPreviewCard)
  description="describes different geometries in the galaxy"
  name="galaxy.geometry"
  namespace="galactic-data"
  physicalName="galaxy.geometry"
  sourceName="redshift"
  tags={
    Array [
      "tag_a",
      "tag_b",
      "tag_c",
    ]
  }
  type="DB_TABLE"
>
  <DatasetPreviewCard
    classes={
      Object {
        "lastUpdated": "DatasetPreviewCard-lastUpdated-2",
        "rightCol": "DatasetPreviewCard-rightCol-1",
      }
    }
    description="describes different geometries in the galaxy"
    name="galaxy.geometry"
    namespace="galactic-data"
    physicalName="galaxy.geometry"
    sourceName="redshift"
    tags={
      Array [
        "tag_a",
        "tag_b",
        "tag_c",
      ]
    }
    type="DB_TABLE"
  >
    <Styled(MuiBox)
      bgcolor="white"
      boxShadow={3}
      display="flex"
      justifyContent="space-between"
      m={1}
      p={2}
    >
      <div
        className="MuiBox-root MuiBox-root-4"
      >
        <div>
          <WithStyles(ForwardRef(Typography))
            color="secondary"
            variant="h3"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              color="secondary"
              variant="h3"
            >
              <h3
                className="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary"
              >
                galaxy.geometry
              </h3>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(WithStyles(ForwardRef(Typography)))
            color="primary"
          >
            <WithStyles(ForwardRef(Typography))
              classes={
                Object {
                  "root": "WithStyles(ForwardRef(Typography))-root-35",
                }
              }
              color="primary"
            >
              <ForwardRef(Typography)
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root WithStyles(ForwardRef(Typography))-root-35",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
                color="primary"
              >
                <p
                  className="MuiTypography-root WithStyles(ForwardRef(Typography))-root-35 MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  describes different geometries in the galaxy
                </p>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(WithStyles(ForwardRef(Typography)))>
        </div>
        <Styled(MuiBox)
          alignItems="flex-end"
          className="DatasetPreviewCard-rightCol-1"
          display="flex"
          flexDirection="column"
          justifyContent="space-between"
        >
          <div
            className="MuiBox-root MuiBox-root-36 DatasetPreviewCard-rightCol-1"
          >
            <div
              id="tagContainer"
            >
              <span
                key="tag_a"
              >
                Pretend Badge
              </span>
              <span
                key="tag_b"
              >
                Pretend Badge
              </span>
            </div>
            <WithStyles(ForwardRef(Typography))
              className="DatasetPreviewCard-lastUpdated-2"
            >
              <ForwardRef(Typography)
                className="DatasetPreviewCard-lastUpdated-2"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
              >
                <p
                  className="MuiTypography-root DatasetPreviewCard-lastUpdated-2 MuiTypography-body1"
                />
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
          </div>
        </Styled(MuiBox)>
      </div>
    </Styled(MuiBox)>
  </DatasetPreviewCard>
</WithStyles(DatasetPreviewCard)>
`;
